kclvm/tools/src/LSP/src/completion.rs

```diff
@@ -225,51 +225,21 @@ fn completion_dot(
     };
 
     if let Some(stmt) = program.pos_to_stmt(&pre_pos) {
-        match stmt.node {
-            Stmt::Import(stmt) => return completion_import(&stmt, &pre_pos, program),
-            _ => {
-                // Todo: string lit has not been processed using the new semantic model and need to handle here.
-                let (expr, _) = inner_most_expr_in_stmt(&stmt.node, &pre_pos, None);
-                if let Some(node) = expr {
-                    if let Expr::StringLit(_) = node.node {
-                        if pre_pos == node.get_end_pos() {
-                            return Some(
-                                into_completion_items(
-                                    &STRING_MEMBER_FUNCTIONS
-                                        .iter()
-                                        .map(|(name, ty)| KCLCompletionItem {
-                                            label: func_ty_complete_label(
-                                                name,
-                                                &ty.into_func_type(),
-                                            ),
-                                            detail: Some(
-                                                ty.into_func_type().func_signature_str(name),
-                                            ),
-                                            documentation: ty.ty_doc(),
-                                            kind: Some(KCLCompletionItemKind::Function),
-                                            insert_text: Some(func_ty_complete_insert_text(
-                                                name,
-                                                &ty.into_func_type(),
-                                            )),
-                                        })
-                                        .collect(),
-                                )
-                                .into(),
-                            );
-                        } else {
-                            return Some(into_completion_items(&items).into());
-                        }
-                    }
-                }
-            }
+        if let Stmt::Import(stmt) = stmt.node {
+            return completion_import(&stmt, &pre_pos, program);
         }
     }
 
     // look_up_exact_symbol
     let mut def = find_def_with_gs(&pre_pos, gs, true);
     if def.is_none() {
-        // look_up_closest_symbol
-        def = find_def_with_gs(pos, gs, false);
+        // handel `symbol?.`
+        let pre_pos = KCLPos {
+            filename: pos.filename.clone(),
+            line: pos.line,
+            column: pos.column.map(|c| c - 2),
```

Comment:
What does `- 2` mean here? Could you pls add more comments here?

Consider the following code

```python
a = "1"?  .capitalize()  # The width of `?  .` is not `2`.
```